mergify_engine/gh_pr_fullifier.py

```diff
@@ -103,78 +103,90 @@ def disabled_by_rules(pull, **extra):
                     % filtered[0])
     return None
 
+# NOTE(sileht): Github mergeable_state is undocumented, here my finding by
+# testing and and some info from other project:
+#
+# unknown: not yet computed by Github
+# dirty: pull request conflict with the base branch
+# behind: head branch is behind the base branch (only if strict: True)
+# unstable: branch up2date (if strict: True) and not required status
+#           checks are failure or pending
+# clean: branch up2date (if strict: True) and all status check OK
+#
+# https://platform.github.community/t/documentation-about-mergeable-state/4259
+# https://github.com/octokit/octokit.net/issues/1763
+
+
+# Use enum.Enum and make is serializable ?
+class MergifyState(object):
```

Comment:
use `enum.Enum`

Reply:
That's planned, but I need first to improve how we serialize/deserialize thing is redis